File: VB/HowToEditAndSavePivotCellValues/Form1.vb
Imports System.Collections.ObjectModel
Imports DevExpress.XtraPivotGrid

Namespace HowToEditAndSavePivotCellValues
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm

		Public Property OrderSourceList() As Collection(Of MyOrderRow)
		Public Sub New()
			InitializeComponent()

			AddHandler pivotGridControl1.CustomCellValue, AddressOf PivotGridControl1_CustomCellValue
			AddHandler pivotGridControl1.EditValueChanged, AddressOf PivotGridControl1_EditValueChanged
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			OrderSourceList = DatabaseHelper.CreateData()
			pivotGridControl1.DataSource = OrderSourceList
			pivotGridControl1.BestFit()
		End Sub

		Private Sub PivotGridControl1_EditValueChanged(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.EditValueChangedEventArgs)
			ChangeCellValue(e, Convert.ToBoolean(e.Value), Convert.ToBoolean(e.Editor.EditValue))
		End Sub

		Private Sub ChangeCellValue(ByVal e As EditValueChangedEventArgs, ByVal oldValue As Boolean, ByVal newValue As Boolean)
			Dim ds As PivotDrillDownDataSource = e.CreateDrillDownDataSource()
			For i As Integer = 0 To ds.RowCount - 1
				ds.SetValue(i, e.DataField, newValue)
			Next i
		End Sub

		Private Sub PivotGridControl1_CustomCellValue(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.PivotCellValueEventArgs)
			If e.DataField.Name = "fieldNeedVerification" AndAlso e.Value IsNot Nothing Then
				e.Value = If(Convert.ToInt32(e.Value) > 0, True, False)
			End If
		End Sub

	End Class
End Namespace

File: CS/HowToEditAndSavePivotCellValues/Form1.cs
using DevExpress.XtraPivotGrid;
using System;
using System.Collections.ObjectModel;

namespace HowToEditAndSavePivotCellValues
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Collection<MyOrderRow> OrderSourceList { get; set; }
        public Form1()
        {
            InitializeComponent();

            pivotGridControl1.CustomCellValue += PivotGridControl1_CustomCellValue;
            pivotGridControl1.EditValueChanged += PivotGridControl1_EditValueChanged;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            OrderSourceList = DatabaseHelper.CreateData();
            pivotGridControl1.DataSource = OrderSourceList;
            pivotGridControl1.BestFit();
        }

        private void PivotGridControl1_EditValueChanged(object sender, DevExpress.XtraPivotGrid.EditValueChangedEventArgs e)
        {
            ChangeCellValue(e, Convert.ToBoolean(e.Value), Convert.ToBoolean(e.Editor.EditValue));
        }

        private void ChangeCellValue(EditValueChangedEventArgs e, bool oldValue, bool newValue)
        {
            PivotDrillDownDataSource ds = e.CreateDrillDownDataSource();
            for (int i = 0; i < ds.RowCount; i++)
            {
                ds.SetValue(i, e.DataField, newValue);
            }
        }

        private void PivotGridControl1_CustomCellValue(object sender, DevExpress.XtraPivotGrid.PivotCellValueEventArgs e)
        {
            if (e.DataField.Name == "fieldNeedVerification" && e.Value != null) {
                e.Value = (Convert.ToInt32(e.Value) > 0) ? true : false;
            }
        }

    }
}
